CoreScriptsRoot/Modules/Server/ServerChat/DefaultChatModules/FriendJoinNotifier.lua

```diff
@@ -0,0 +1,47 @@
+--	// FileName: FriendJoinNotifer.lua
+--	// Written by: TheGamer101
+--	// Description: Module that adds a message to the chat whenever a friend joins the game.
+
+local Chat = game:GetService("Chat")
+local Players = game:GetService("Players")
+
+local ReplicatedModules = Chat:WaitForChild("ClientChatModules")
+local ChatSettings = require(ReplicatedModules:WaitForChild("ChatSettings"))
+local ChatConstants = require(ReplicatedModules:WaitForChild("ChatConstants"))
+
+local FriendMessageTextColor = Color3.fromRGB(255, 255, 255)
+local FriendMessageExtraData = {ChatColor = FriendMessageTextColor}
+
+local function Run(ChatService)
+
+	local function ShowFriendJoinNotification()
+		if ChatSettings.ShowFriendJoinNotification ~= nil then
+			return ChatSettings.ShowFriendJoinNotification
+		end
+		return false
+	end
+
+	local function SendFriendJoinNotification(player, joinedFriend)
+		local speakerObj = ChatService:GetSpeaker(player.Name)
+		if speakerObj then
+			speakerObj:SendSystemMessage(string.format("Your friend %s has joined the game.", joinedFriend.Name), "System", FriendMessageExtraData)
+		end
+	end
+
+	if ShowFriendJoinNotification() then
+		Players.PlayerAdded:connect(function(player)
```

Comment:
Why are you scanning every player for friends? I do believe that you could do this instead:
```lua
Players.PlayerAdded:connect(function(player)
     if player:IsFriendsWith(Players.LocalPlayer.UserId) then
          SendFriendJoinNotification(player, Players.LocalPlayer)
     end
end